### ISSUE TITLE
build: installer migration related cleanups

### ISSUE THREAD

**Issue by fichtner** [opnsense/installer#7]
- [x] .abi_hint seems to be present in the mtree file for installation media https://github.com/opnsense/tools/commit/4a8970cbf8
- [x] Default to GPT instead of MBR to align with our changes from 2016 https://github.com/opnsense/src/commit/66421e9dae
- [x] Default to GPT/UEFI instead of autodetecting boot method from the system to align with our changes from 2016 97c9789818b09 dc78145c
- [x] no GPT default labels (bootfs/rootfs/swapfs) are added in the setup and used in /etc/fstab 79e3c4c389c46 f4e2531a9cb8a 5a023c907e085
- [ ] ~~machdep.bootmethod is being queried for defaults all the time making it impossible to use GPT with UEFI default when not having booted from UEFI  -- UEFI is supposed to be backwards compatible and we have been using that fine since 2016 too~~ we are not going to use that code, at least not for our autoinstallers
- [ ] ~~partition layout for opnsense-importer changes with bsdinstall~~ having fixed UEFI hybrid boot this works as expected
- [ ] ~~move installer user to integrated authentication framework~~ since the installer user is only found on the disk we do need the password hash in the local UNIX account because integrated authentication will try to find it in the local config.xml database only

Basically now we worked around the bsdinstaller autoinstall problems going back to what we had in the bsdinstaller and not being able to use ZFS without further work.  ¯\_(ツ)_/¯
